File: ssm-sdk-client/src/main/java/ssm/client/invoke/query/AbstractQuery.java
package ssm.client.invoke.query;

import ssm.client.invoke.command.InvokeArgs;

public abstract class AbstractQuery {

    public abstract String functionGetValue();

    public InvokeArgs queryArgs(String username) {
        return new InvokeArgs(functionGetValue(), username);
    }

    public InvokeArgs listArgs() {
        return new InvokeArgs(HasList.LIST_FUNCTION, functionGetValue());
    }
}

File: ssm-chaincode/ssm-chaincode-client/src/main/kotlin/ssm/chaincode/client/invoke/query/AbstractQuery.kt
package ssm.chaincode.client.invoke.query

import ssm.chaincode.client.invoke.command.InvokeArgs

abstract class AbstractQuery {
	abstract fun functionGetValue(): String
	fun queryArgs(username: String): InvokeArgs {
		return InvokeArgs(functionGetValue(), listOf(username))
	}

	fun listArgs(): InvokeArgs {
		return InvokeArgs(HasList.LIST_FUNCTION, listOf(functionGetValue()))
	}
}
